backend/app.go

```diff
@@ -53,73 +47,131 @@ func getPrograms(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	respondWithJSON(w, http.StatusOK, payload)
 }
 
+func runProfiling(packageName string) {
+	profilingisDone := false
+
+	if !profilingStarted {
+		profilingStarted = true
+		go func() {
+			profiler.Profiler(packageName, profilingDoneChannel)
+		}()
+
+		for !profilingisDone {
+			cpuStats := profiler.CPUPercent()
+			graphPoints["Time"] = append(graphPoints["Time"], cpuStats[0])
+			graphPoints["Percent"] = append(graphPoints["Percent"], cpuStats[1])
+
+			timer := time.NewTimer(50 * time.Millisecond)
+			<-timer.C
+			profilingisDone = checkIfProfilingisDone()
+		}
+	}
+}
+
+func checkIfPprofFileExists() bool {
+	if _, err := os.Stat("cpu.pprof"); os.IsNotExist(err) {
+		fmt.Println("pprof does not exists")
+		return false
+	}
+	fmt.Println("pprof exists")
+	return true
+}
+
+func checkIfDiagramExists(packageName string) bool {
+	if _, err := os.Stat("./diagrams/" + packageName); os.IsNotExist(err) {
+		fmt.Println("diagram does not exists")
+		return false
+	}
+	fmt.Println("diagram exists")
+	return true
+}
+
+func checkIfReportExists(packageName string) bool {
```

Comment:
Very similar to `checkIfDiagramExists`. What if `/diagrams/` and `/reports/` is part of the URL?